src/platform/nm-linux-platform.c

```diff
@@ -2019,8 +2019,21 @@ _wireguard_update_from_peers_nla (CList *peers,
 
 		if (tb[WGPEER_A_PERSISTENT_KEEPALIVE_INTERVAL])
 			peer_c->data.persistent_keepalive_interval = nla_get_u16 (tb[WGPEER_A_PERSISTENT_KEEPALIVE_INTERVAL]);
-		if (tb[WGPEER_A_LAST_HANDSHAKE_TIME])
-			nla_memcpy (&peer_c->data.last_handshake_time, tb[WGPEER_A_LAST_HANDSHAKE_TIME], sizeof (peer_c->data.last_handshake_time));
+		if (tb[WGPEER_A_LAST_HANDSHAKE_TIME]) {
+			if (nla_len (tb[WGPEER_A_LAST_HANDSHAKE_TIME]) >= sizeof (peer_c->data.last_handshake_time))
+				nla_memcpy (&peer_c->data.last_handshake_time, tb[WGPEER_A_LAST_HANDSHAKE_TIME], sizeof (peer_c->data.last_handshake_time));
+			else if (   sizeof (struct timespec) < sizeof (NMPTimespec64)
+			         && nla_len (tb[WGPEER_A_LAST_HANDSHAKE_TIME]) >= sizeof (struct timespec)) {
+				struct timespec ts;
+
+				/* older versions used "struct timespec". */
```

Comment:
Personally, I'd prefer you don't leave support for older code around.

Reply:
OK, dropped (also, because the value is unused and because the code only mattered on 32bit). 

While at it, I think it's inconvenient to expose timestamps on netlink in `time(2)` scale, which is susceptible to clock changes. 
 I think `CLOCK_BOOTTIME` scale would be better, otherwise there is an unresolvable race computing `ago()`.